oauth-common/src/main/java/io/strimzi/kafka/oauth/services/OAuthMetrics.java

```diff
@@ -0,0 +1,179 @@
+
+    private final Map<MetricKey, Sensor> sensorMap = new ConcurrentHashMap<>();
+
+    OAuthMetrics(Map<String, ?> configMap) {
+        this.configMap = configMap;
+        this.config = new Config(configMap);
+        this.metrics = initKafkaMetrics();
+    }
+
+    public void addTime(MetricKey key, long time) {
+        Sensor registeredSensor = sensorMap.computeIfAbsent(key, k -> {
+            Sensor sensor = metrics.sensor(key.getMBeanName());
+            addMetricsToSensor(metrics, sensor, key);
+            return sensor;
+        });
+
+        registeredSensor.record(time);
+    }
+
+    private Metrics initKafkaMetrics() {
+        List<MetricsReporter> reporters = initReporters();
+        KafkaMetricsContext ctx = createKafkaMetricsContext();
+
+        MetricConfig metricConfig = getMetricConfig();
+
+        return createMetrics(metricConfig, ctx, reporters);
+    }
+
+    private List<MetricsReporter> initReporters() {
+        AbstractConfig kafkaConfig = initKafkaConfig();
+        List<MetricsReporter> reporters = kafkaConfig.getConfiguredInstances(CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
+                MetricsReporter.class);
+
+        JmxReporter reporter = new JmxReporter();
+        reporter.configure(configMap);
+
+        reporters.add(reporter);
+        return reporters;
+    }
+
+    private AbstractConfig initKafkaConfig() {
+        ConfigDef configDef = new ConfigDef()
+                .define(CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
+                        ConfigDef.Type.LIST,
+                        Collections.emptyList(),
+                        new ConfigDef.NonNullValidator(),
+                        ConfigDef.Importance.LOW,
+                        CommonClientConfigs.METRIC_REPORTER_CLASSES_DOC);
+
+        return new AbstractConfig(configDef, configMap);
+    }
+
+    private KafkaMetricsContext createKafkaMetricsContext() {
+        HashMap<String, String> contextLabels = new HashMap<>();
+
+        // We don't know how to get clusterId here
+        //contextLabels.put("kafka.cluster.id", getClusterId());
+
+        // Broker side init code (for reference)
+        //if (config.usesSelfManagedQuorum) {
+        //    contextLabels.put("kafka.node.id", nodeId)
+        //} else {
+        //    contextLabels.put("kafka.broker.id", brokerId)
+        //}
+
+        String processRoles = config.getValue("process.roles");
+        if (processRoles == null || "[]".equals(processRoles)) {
+            String brokerId = config.getValue("broker.id");
+            if (brokerId != null) {
+                contextLabels.put("kafka.broker.id", brokerId);
+            }
+        } else {
+            String nodeId = config.getValue("node.id");
+            if (nodeId != null) {
+                contextLabels.put("kafka.node.id", nodeId);
+            }
+        }
+
+        if (config.getValue("client.id") != null) {
+            String clientId = config.getValue("client.id");
```

Comment:
Should we use `CommonClientConfigs.CLIENT_ID_CONFIG` instead of `client.id`?

Reply:
Done.